### ISSUE TITLE
Error in Rule "Rule0017" of type "InvalidCastException"

### ISSUE THREAD

**Issue by fvet** [StefanMaron/BusinessCentral.LinterCop#164]
Code below results into this warning:

_There was an Error in Rule "Rule0017" of type "InvalidCastException" at Line ..._

![image](https://user-images.githubusercontent.com/25268332/155892988-f890c687-46dc-49ef-9cad-7b89ac2d776b.png)

```
local procedure ParseDictionaryToText(Dictionary: Dictionary of [Text, Text]) Values: text;
    var
        ListOfKeys: list of [text];
        Loop: Integer;
        CRLF: text;
    begin
        CRLF[1] := 13;
        CRLF[2] := 10;

        ListOfKeys := Dictionary.Keys();

        for Loop := 1 to ListOfKeys.Count() do begin
            if Values <> '' then Values += CRLF;
            Values += ListOfKeys.get(Loop) + ': ' + Dictionary.Get(ListOfKeys.get(Loop));
        end;
    end;
```

**Comment by StefanMaron** [comment]
Should be fixed in https://github.com/StefanMaron/BusinessCentral.LinterCop/releases/tag/v0.24.8
Compare #181

**Comment by fvet** [comment]
Switched back to a 8.4.x compiler and in that version the info disappeared. Thx.
Not sure if it's allowed to test LC (pre-)releases in combination with AL insider vsix.

**Comment by StefanMaron** [comment]
I am working on a way to (auto) ship a linter dll which is compiled against the next major runtime version. But right now, I have no way to do this. 

But please, by all means, feel free to test this against the next version as well. I introduced a new label to mark issues that appear with the next major version. And I do test this locally to make sure the next major version works. 

To give a little background why these bugs appear:
With major releases MS releases new versions of the runtime and because they do not officially support building custom analyzers (yet) they do not care about braking changes. So when they add new values for a property for example, the internal IDs of those properties sometimes "shift"  so for example in Runtime 8 fieldclass flowfield had an internal id of 3, and in runtime version 9 fieldclass flowfilter points to id 3. When you then use the dll compiled against runtime 8, only that id is referenced not the term "flowfield" and that causes that my code, which meant to point to flowfield, will point to flowfilter with runtime 9

Long story short, those bugs get fixed if I just do a recompile against the new runtime version, and I will do this and release the new dll as soon as the new runtime is published on the vscode marketplace
